docs/serialization.rst

```diff
@@ -33,46 +34,30 @@ as training. Techncially there are some difficulties with this approach:
 * We do not want to serialize the training data kept in memory, since this can
   be prohibitively large.
 
-Blocks addresses these problems by using a pickling extension called Dill,
-avoiding certain data structures such as generators and nested classes (see the
-:ref:`developer guidelines <serialization_guidelines>`), and by overriding the
-pickling behavior of datasets.
+Blocks addresses these problems by avoiding certain data structures such as
+generators and nested classes (see the :ref:`developer guidelines
+<serialization_guidelines>`) and overriding the pickling behaviour of some
+objects, making the pickling of the main loop possible.
 
-However, in general you should not rely on this serialization approach for the
-long term saving of your models. Problems that remain are
+However, pickling can be problematic for long-term storage of models, because
 
 * Unpickling depends on the libraries used being unchanged. This means that if
   you updated Blocks, Theano, etc. to a new version where the interface has
-  changed, loading your training progress will fail.
+  changed, loading your training progress could fail.
 * The unpickling of Theano objects can be problematic, especially when
   transferring from GPU to CPU or vice versa.
 * It is not possible on Python 2 to unpickle objects that were pickled in Python
   3.
 
-.. note::
-
-   On the long term, we plan to serialize the log, data stream, and the rest of
-   the main loop separately. This way you can e.g. perform plotting without
-   needing to deserialize the Theano model.
-
 Parameter saving
 ----------------
 
-The second method used by Blocks is intended to be more cross-platform, and a
-safer way of storing models for a longer period of time. This method:
-
-* Stores the parameters in a binary NumPy file (``.npz``)
-* Serializes the log
-* Serializes the data stream
-
-When resuming training, the model is reconstructed after which the parameters
-can be reloaded from the NumPy file. The training log and data stream are loaded
-as well, allowing the training to continue. However, this method makes no effort
-to try and store the exact state of training. This means that:
+This is why Blocks intercepts the pickling of all Theano shared variables (which
+includes the parameters), and stores them as separate NPY_ files. The resulting
+file is a ZIP arcive that contains the pickled main loop as well as a collection
+of NumPy arrays. The NumPy arrays (and hence parameters) in the ZIP file can be
+read, across platforms, using the :func:`numpy.load` function, making it
+possible to inspect and load parameter values, even if the unpickling of the
+main loop fails.
```

Comment:
Yeah, although I would put that in the docstring, since it's more low-level stuff. I intended this file to be more of a high-level overview of serialization in Blocks (what methods are used, what shortcomings are addressed, etc.).